### PR TITLE
Add generated code docs as docs/index.md

### PR DESCRIPTION
**Pull request by dibenede**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+Closure style, you run the compiler with the `import_style=commonjs` option. The
+names of the output files are computed by taking the name of the each input
+`.proto` file and making two changes.
+
+**Note:** Specifying a `library` option is ignored with this import style.
+
+So, for example, let's say you invoke the compiler as follows:
+
+```
+protoc --proto_path=src --js_out=import_style=commonjs,binary:build/gen src/foo.proto src/bar/baz.proto
```
Comment (lukesandberg) [inline docs/index.md:62]: are there additional params we need to pass for the plugin

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+build it first by running:
+
+```
+PROTOC=/path/to/protoc PROTOC_INC=/path/to/proto/include gulp dist
+```
+
+You should be able to import your generated types with statements like:
+
+```js
+var messages = require('./messages_pb');
```
Comment (lukesandberg) [inline docs/index.md:83]: nit: lets update these vars to const like its 2015

**Review by lukesandberg — APPROVED** [review]

File: docs/index.md (hunk 1 above)
Comment (dibenede) [reply docs/index.md]: Added --plugin to be explicit.  Not sure if there was some intended context in the above command, but the rest seems to be generally ok per my verification attempts.

File: docs/index.md (hunk 2 above)
Comment (dibenede) [reply docs/index.md]: done

**Pull request merged**